include/swift/AST/DiagnosticsSIL.def

```diff
@@ -371,6 +371,23 @@ ERROR(possible_return_before_yield, none,
 
 NOTE(one_yield, none, "yield along one path is here", ())
 
+NOTE(false_branch_doesnt_yield, none,
+    "must yield when the condition is false", ())
+
+NOTE(true_branch_doesnt_yield, none,
+    "must yield when the condition is true", ())
+
+NOTE(switch_case_doesnt_yield, none, "must yield in the case '%0'", (StringRef))
```

Comment:
I would reorder this to: "in the '%0' case". Also, if you use an identifier here rather than a StringRef then the diagnostics system will put in single quotes for you.